var/spack/repos/builtin/packages/charliecloud/package.py

```diff
@@ -33,6 +34,10 @@ class Charliecloud(AutotoolsPackage):
     depends_on('py-sphinx',           type='build', when='+docs')
     depends_on('py-sphinx-rtd-theme', type='build', when='+docs')
 
+    if sys.platform == 'darwin':
+       raise InstallError("This package does not build on Mac OS X\n"
+                          "see: https://github.com/hpc/charliecloud/issues/42")
```

Comment:
```suggestion
    conflicts('platform=darwin', msg='This package does not build on macOS')
```
The previous logic will break Spack as the class needs to be loaded in order for Spack to concretize any package, not just Charliecloud.